lib/taurus/core/tango/tangodevice.py

```diff
@@ -185,9 +185,14 @@ def getDisplayDescrObj(self, cache=True):
         ret = []
         for name, value in desc_obj:
             if name.lower() == u'device state' and self.stateObj is not None:
-                tg_state = self.stateObj.read(cache).rvalue.name
-                value = u"%s (%s)" % (value, tg_state)
-            ret.append((name, value))
+                try:
```

Comment:
I would prefer to make this exception handling a bit simpler and more robust (even if sacrificing some info in the tooltip). I propose  to replace the try-except block by:

```python
    try:
        tg_state = self.stateObj.read(cache).rvalue.name
    except Exception as e:
        tg_state = "cannot read state"
    value = u"%s (%s)" % (value, tg_state)
```

Reply:
I prefer not to hide any exception.

For example using `Expcetion` will hide typo in the code. And the content of `DevFail` can be important. But i guess it there is an exception there, it will also be catched somewhere else in the code.

But while it works, i am fine with your proposal.